workflow-examples/src/test/java/com/redhat/parodos/examples/integration/utils/ExamplesUtils.java

```diff
@@ -26,34 +27,97 @@
 public final class ExamplesUtils {
 
 	public static void waitProjectStart(ProjectApi projectApi) throws ApiException, InterruptedException {
-		AsyncResult asyncResult = new AsyncResult();
+		AsyncWaitProjectResult asyncResult = new AsyncWaitProjectResult();
 		Lock lock = new ReentrantLock();
 		Condition response = lock.newCondition();
 		ApiCallback<List<ProjectResponseDTO>> apiCallback = new ApiCallback<>() {
-			AtomicInteger failureCounter = new AtomicInteger(0);
+			@Override
+			public void onFailure(ApiException e, int statusCode, Map<String, List<String>> responseHeaders) {
+				try {
+					projectApi.getProjectsAsync(this);
+				}
+				catch (ApiException apie) {
+					asyncResult.setError(apie.getMessage());
+					signal();
+				}
+			}
+
+			@Override
+			public void onSuccess(List<ProjectResponseDTO> result, int statusCode,
+					Map<String, List<String>> responseHeaders) {
+				signal();
+			}
+
+			@Override
+			public void onUploadProgress(long bytesWritten, long contentLength, boolean done) {
+			}
+
+			@Override
+			public void onDownloadProgress(long bytesRead, long contentLength, boolean done) {
+			}
+
+			private void signal() {
+				lock.lock();
+				try {
+					response.signal();
+				}
+				finally {
+					lock.unlock();
+				}
+			}
+		};
+		projectApi.getProjectsAsync(apiCallback);
+		lock.lock();
+		try {
+			// should be more than enough
+			response.await(60, TimeUnit.SECONDS);
+			if (asyncResult.getError() != null) {
+				fail("An error occurred while executing getProjectAsync: " + asyncResult.getError());
+			}
+		}
+		finally {
+			lock.unlock();
+		}
+	}
+
+	public static WorkFlowStatusResponseDTO waitAsyncStatusResponse(WorkflowApi workflowApi, String workFlowExecutionId)
```

Comment:
do we want to unify this method and `waitProjectStart`?

Reply:
@pkliczewski done! Can you please take a look?